### ISSUE TITLE
Ernie-M for pretraining multilingual models

### ISSUE THREAD

**Issue by KnutJaegersberg** [huggingface/transformers#22257]
### Feature request

Two things that might help in that regard: 

- To train TSDAE, one needs support as class ErnieMForPreTraining, just as for Ernie
https://huggingface.co/docs/transformers/model_doc/ernie#transformers.ErnieForPreTraining

- To train cross-encoders with contrastive loss, a bit like SimCSE, one needs standard support for getting the 'attention_mask' out of the tokenizer sbert uses. Sbert just expects those. Tried to hack it in into sbert, but failed. 

### Motivation

Suspect getting Ernie-M-large for pretraining multilingual sentence embeddings will yield close to sota results.  
According to mSimCSE, we can get top multilingual embeddings just on training on their 300k dataset of english pairs, alone (worked better than cross-lingual training). 
With a stronger base model (they used xlm-roberta), sota embeddings might just lie on the streets.  
https://github.com/yaushian/mSimCSE

### Your contribution

Can't do it alone, plz help.

**Comment by amyeroberts** [comment]
Hi @KnutJaegersberg, thanks for making this suggestion! 

Would you like to try and open a PR to add the model? We have guidance written on adding [models here](https://huggingface.co/docs/transformers/v4.27.2/en/add_new_model). As the modeling file already exists - adding this component is even easier than a whole new model. For example, see [this PR](https://github.com/huggingface/transformers/pull/21754) for adding `WhisperForAudioClassification`.

**Comment by KnutJaegersberg** [comment]
Currently in the middle of something, will try to look at it later!

**Comment by pruksmhc** [comment]
I'd love to try this out!

**Comment by pruksmhc** [comment]
Seems like this will require more than simply copy-pasting the BertForPretraining code, but actually implementing cross-attention Masked Language Modeling and Back-translation Masked Language Modeling.

**Comment by qgallouedec** [comment]
Hey @KnutJaegersberg, sorry for the very slow reply, we're catching up on the backlog. Ernie-M was deprecated for low usage back in 2024 and has since been removed from the library entirely (#42235), so there's no more `ErnieMModel` to build `ErnieMForPreTraining` on top of.

Closing as outdated. If it helps, plain Ernie already ships `ErnieForPreTraining`:

https://github.com/huggingface/transformers/blob/4da05482135896a529d5536c3c003102d36528a2/src/transformers/models/ernie/modeling_ernie.py#L735

and for multilingual sentence embeddings today, XLM-RoBERTa or other current encoder models plus the sentence-transformers training recipes are the better starting point.
